### DISCUSSION TITLE
Constraint driven design/generation of geometry

### DISCUSSION THREAD

**Discussion by JimJoseStudent** [leap71/PicoGK#34] [Q&A]
Hi, 
i have often noticed that there is a consideration for performance driven generation of geometry discussed in some models. If im not mistaken it is to mean that lets say we are designing a rocket engine, and we want to generate a cooling channel within the wall, if i want  it to be designed based on say a requirement of 5m/s flow rate it is possible?

if so would you guys happen to have one example of such an application?

**Comment by LinKayser** [comment]
Yes, this is how Noyron works, our internal Large Computational Engineering Model. So it takes these parameters into account, and, for example varies the cross-section of a Volute, etc. Same with thermal models, which actively calculate cooling needs, and then add fins, increase cooling channel cross section, etc.

Below is a good example of a volute that changes cross section, based on the actual fluid mass at each point (the pipes remove parts of the mass, so the cross section gets smaller).

![4CC20CD7-46C5-43A7-8D34-0E64648A6563](https://github.com/leap71/PicoGK/assets/148448288/e8b305c8-ec02-4bdc-aa0a-5ba57727c92d)

PicoGK, the geometry kernel, has no such functionality, so you have to design these rules yourself. While we are considering open-sourcing parts of Noyron, at this time it's all closed source and only open to our customers.

My [book will slowly move towards](https://picogk.org/coding-for-engineers/) the implementation of these features. So with a bit of patience, you should get a good background.

Best,
Lin

**Comment by JimJoseStudent** [comment]
HI Lin, 

Sorry for the late response. that answeres my question and looking forward to the rest of your book. It is quiet helpful and has helped me a lot already. 


> sourcing parts of Noyron, at this time it's all closed source and only open to our customers.

makes sense.
